File: packages/docs-test/tests/typescript/quickstart.ts
import "dotenv/config"
import Supermemory from "supermemory"

const client = new Supermemory()
const USER_ID = "docs-test-user"

const conversation = [
	{ role: "assistant", content: "Hello, how are you doing?" },
	{
		role: "user",
		content: "Hello! I am Dhravya. I am 20 years old. I love to code!",
	},
	{ role: "user", content: "Can I go to the club?" },
]

async function main() {
	console.log("Testing quickstart TypeScript code...\n")

	// Get user profile + relevant memories for context
	console.log("1. Getting user profile...")
	const profile = await client.profile({
		containerTag: USER_ID,
		q: conversation.at(-1)!.content,
	})

	console.log("Profile response:", JSON.stringify(profile, null, 2))

	const context = `Static profile:
${profile.profile.static.join("\n")}

Dynamic profile:
${profile.profile.dynamic.join("\n")}

Relevant memories:
${profile.searchResults?.results.map((r) => r.content).join("\n")}`

	console.log("\n2. Built context:", context)

	// Build messages with memory-enriched context
	const messages = [
		{ role: "system", content: `User context:\n${context}` },
		...conversation,
	]
	console.log("\n3. Messages built successfully")

	// Store conversation for future context
	console.log("\n4. Adding memory...")
	const addResult = await client.add({
		content: conversation.map((m) => `${m.role}: ${m.content}`).join("\n"),
		containerTag: USER_ID,
	})

	console.log("Add result:", JSON.stringify(addResult, null, 2))
	console.log("\n✅ Quickstart TypeScript test passed!")
}

main().catch(console.error)

File: packages/docs-test/tests/python/quickstart.py
import os
from dotenv import load_dotenv

load_dotenv()

from supermemory import Supermemory

client = Supermemory()
USER_ID = "docs-test-user-py"

conversation = [
    {"role": "assistant", "content": "Hello, how are you doing?"},
    {"role": "user", "content": "Hello! I am Dhravya. I am 20 years old. I love to code!"},
    {"role": "user", "content": "Can I go to the club?"},
]

print("Testing quickstart Python code...\n")

# Get user profile + relevant memories for context
print("1. Getting user profile...")
profile = client.profile(container_tag=USER_ID, q=conversation[-1]["content"])

print(f"Profile response: {profile}")

def get_memory(r):
    if hasattr(r, 'memory'):
        return r.memory
    return r.get('memory', '') if isinstance(r, dict) else str(r)

context = f"""Static profile:
{chr(10).join(profile.profile.static)}

Dynamic profile:
{chr(10).join(profile.profile.dynamic)}

Relevant memories:
{chr(10).join(get_memory(r) for r in profile.search_results.results)}"""

print(f"\n2. Built context: {context}")

# Build messages with memory-enriched context
messages = [{"role": "system", "content": f"User context:\n{context}"}, *conversation]
print("\n3. Messages built successfully")

# Store conversation for future context
print("\n4. Adding memory...")
add_result = client.add(
    content="\n".join(f"{m['role']}: {m['content']}" for m in conversation),
    container_tag=USER_ID,
)

print(f"Add result: {add_result}")
print("\n✅ Quickstart Python test passed!")
